pyphare/pyphare/pharesee/geometry.py

```diff
@@ -0,0 +1,277 @@
+from ..core import box as boxm
+from .hierarchy import FieldData, is_root_lvl
+
+
+
+def toFieldBox(box, patch_data):
+    """
+    grows the box by one cell if patch_data is primal
+    because box.upper+1 allows to get the rightmost primal node
+    otherwise missing
+    """
+    assert (isinstance(patch_data, FieldData))
+    if patch_data.layout.centering["X"][patch_data.field_name] == "primal":
+        return boxm.Box(box.lower, box.upper + 1)
+    else:
+        return box
+
+
+
+
+
+def shift_patch(patch, offset):
+    patch.box = boxm.shift(patch.box, offset)
+    for pdata in patch.patch_datas.values():
+        pdata.ghost_box = boxm.shift(pdata.ghost_box, offset)
+
+
+
+
+def touch_domain_border(box, domain_box, border):
+    if border == "upper":
+        return domain_box.upper in box
+    elif border == "lower":
+        return domain_box.lower in box
+    else:
+        raise RuntimeError("invalid border")
+
+
+
+def compute_overlaps(patches, domain_box):
+    """
+    returns a list of overlaps for all patch datas in given patches
+    and for a domain box. An overlap is defined as an intersection of
+    two PatchData ghost box.
+
+    if  - = domain cell, o = ghost cell
+
+    PatchData1  : - - - - - - - - - o o o o o
+    PatchData2  :                     o o o o o - - - - - - - -
+    overlap box :                     ^ ^ ^ ^
+
+    for now the function is purely 1D and assumes the direction is
+    periodic. Therefore, PatchData close to the lower (resp. upper)
+    domain boundary (hence the need of domain_box) may overlap another
+    one close to the upper (resp. lower) end of the domain.
+
```

Comment:
this is details, but the cartoon would refer to left and right, while it is lower and upper in the explanation

Reply:
well lower and upper is dimension independent. left is lower of right and right is upper of left, but left/right is often considered to be on horizontal axis... lower/upper is more general. a box for instance is defined by two points, lower and upper.... no matter the dimension...